scripts/dev/change_version.py

```diff
@@ -76,49 +73,70 @@ def usage():
 
 
 # check the command line argument status
-if not len(sys.argv) == 4:
+if len(sys.argv) != 4:
     print("Invalid command line arguments")
     usage()
     sys.exit(1)
 
 # store the command line arguments since they appear valid so far
-repo = sys.argv[1]
+repo = Path(sys.argv[1]).resolve()
 v_old = sys.argv[2]
 v_new = sys.argv[3]
 
+# folders to search
+idf_folders = [
+    Path("testfiles"),
+    Path("performance_tests"),
+    Path("datasets"),
+    Path("testfiles") / "BasicsFiles",
+]
+
 # walk across the idf and imf files
-for extension in ["*.idf", "*.imf"]:
-    for folder in ["testfiles", "performance_tests", "datasets", os.path.join("testfiles", "BasicsFiles")]:
-        this_dir = os.path.join(repo, folder)
-        for root, dir_names, file_names in os.walk(this_dir):
-            for filename in fnmatch.filter(file_names, extension):
-                with codecs.open(os.path.join(root, filename), encoding="utf-8", errors="ignore") as input_file:
```

Comment:
`codecs.open` was warning as deprecated, so I updated that. Also updated the rest of the script to use `pathlib` everywhere.